src/journal/journal-file.c

```diff
@@ -274,6 +312,28 @@ int journal_file_set_offline(JournalFile *f, bool wait) {
         return 0;
 }
 
+/* Calls fsync on the underlying fd, respecting and maintaining f->fsync_error as needed.
+ * Not intended to be called from the offlining thread as f->fsync_error access is unserialized
+ */
+static int journal_file_fsync(JournalFile *f) {
+        int r;
+
+        assert(f);
+        assert(f->fd >= 0);
+
+        if (f->fsync_error < 0)
+                return f->fsync_error;
+
+        r = fsync(f->fd);
+        if (r < 0) {
```

Comment:
I don't think `r` is useful at all, plain `if (fsync(f->fd) <  0)` works just as well.